streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/ClientState.java

```diff
@@ -59,28 +53,92 @@ private ClientState(Set<T> activeTasks, Set<T> standbyTasks, Set<T> assignedTask
     }
 
     public void assign(T taskId, boolean active) {
-        if (active)
+        if (active) {
             activeTasks.add(taskId);
-        else
+        } else {
             standbyTasks.add(taskId);
+        }
 
         assignedTasks.add(taskId);
+    }
+
+    public Set<T> activeTasks() {
+        return activeTasks;
+    }
+
+    public Set<T> standbyTasks() {
+        return standbyTasks;
+    }
 
-        double cost = COST_LOAD;
-        cost = prevAssignedTasks.remove(taskId) ? COST_STANDBY : cost;
-        cost = prevActiveTasks.remove(taskId) ? COST_ACTIVE : cost;
+    public int assignedTaskCount() {
+        return assignedTasks.size();
+    }
+
+    public void incrementCapacity() {
+        capacity++;
+    }
+
+    public int activeTaskCount() {
+        return activeTasks.size();
+    }
+
+    public void addPreviousActiveTasks(final Set<T> prevTasks) {
+        prevActiveTasks.addAll(prevTasks);
+        prevAssignedTasks.addAll(prevTasks);
+    }
 
-        this.cost += cost;
+    public void addPreviousStandbyTasks(final Set<T> standbyTasks) {
+        prevAssignedTasks.addAll(standbyTasks);
     }
 
     @Override
     public String toString() {
         return "[activeTasks: (" + activeTasks +
             ") assignedTasks: (" + assignedTasks +
+            ") standbyTasks: (" + standbyTasks +
             ") prevActiveTasks: (" + prevActiveTasks +
             ") prevAssignedTasks: (" + prevAssignedTasks +
             ") capacity: " + capacity +
-            " cost: " + cost +
             "]";
     }
+
+    boolean reachedCapacity() {
+        return assignedTasks.size() >= capacity;
```

Comment:
@dguy This is related to the general comment I had above: do we still try to favor stickiness after each thread has assigned a task? In my general comment I asked for an example, and I'm thinking now about another slightly different one: 

Say we have two client with one thread (hence one capacity) each and four tasks, and each client is assigned two tasks `(1, 2), (3, 4)`, both of them are "over-capacity" now. When adding a new client with one thread, one of the task will be migrated to that client, say task4. The question is with this assignor, do we enforce the assignment to be `(1, 2), (3), (4)` or task2 can be randomly assigned to another client after task1 has been assigned to client1?